File: track/core/helpers/time_helper.py
from datetime import datetime, timedelta


class TimeHelper:
    @staticmethod
    def get_current_utc_date():
        return datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%S") + "+00:00"

    @staticmethod
    def seconds_to_hms_string(seconds):
        hours, remainder = divmod(seconds, 3600)
        minutes, seconds = divmod(remainder, 60)
        return f"{TimeHelper._add_zero_prefix(hours)}h {TimeHelper._add_zero_prefix(minutes)}m {TimeHelper._add_zero_prefix(seconds)}s"

    @staticmethod
    def _add_zero_prefix(number):
        return number if number > 9 else f"0{number}"

    @staticmethod
    def get_week_dates(date):
        day_of_week = date.weekday()
        start_of_week = date - timedelta(days=day_of_week)
        end_of_week = start_of_week + timedelta(days=6)

        return start_of_week, end_of_week

File: src/core/entities/time-helper.ts
export class TimeHelper {
  static secondsToHmsString(seconds: number): string {
    const hours = Math.floor(seconds / 3600);
    const minutes = Math.floor((seconds % 3600) / 60);
    const remainingSeconds = seconds % 60;
    return `${TimeHelper.addZeroPrefix(hours)}h ${TimeHelper.addZeroPrefix(minutes)}m ${TimeHelper.addZeroPrefix(remainingSeconds)}s`;
  }

  private static addZeroPrefix(number: number): string {
    return number > 9 ? number.toString() : `0${number}`;
  }

  getCurrentUtcDate(): string {
    const now = new Date();
    return now.toISOString();
  }

  getWeekDates(date: Date): [Date, Date] {
    const dayOfWeek = date.getDay() === 0 ? 6 : date.getDay() - 1

    const startOfWeek = new Date(date);
    startOfWeek.setDate(date.getDate() - dayOfWeek);

    const endOfWeek = new Date(startOfWeek);
    endOfWeek.setDate(startOfWeek.getDate() + 6);

    return [startOfWeek, endOfWeek];
  }
}
